packages/cinterop/src/commonMain/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -539,7 +539,7 @@ expect object RealmInterop {
     fun realm_user_refresh_custom_data(app: RealmAppPointer, user: RealmUserPointer, callback: AppCallback<Unit>)
 
     // Sync client config
-    fun realm_sync_client_config_new(): RealmSyncClientConfigurationPointer
+    fun realm_sync_client_config_new(appId: String): RealmSyncClientConfigurationPointer
```

Comment:
Maybe add a comment about why we need the `appId` here ... but I guess the need for that ties into my comment against actually registering the thread-notifier-callback in here.